lib/chat/actions.tsx

```diff
@@ -191,6 +191,15 @@ Besides that, you can also chat with users and do some calculations if needed.`
       }
 
       if (done) {
+        analytics.track({
+          userId: '123',
+          event: 'Message Received - Server',
+          properties: {
+            content,
+            conversationId: aiState.get().chatId
+          }
+        })
+
```

Comment:
Ideally we can track Message Received on the server instead of client to avoid problems with duplicate messages due to component remounting